### ISSUE TITLE
Wrong WAL arithmetic with non-default WAL segment size

### ISSUE THREAD

**Issue by gbartolini** [2ndquadrant-it/barman#287]
Reported by Domagoj Smoljanovic in the mailing list, see original message below.

Recently we created one server with different wal segment size (64MB) making WALs not follow the naming pattern of 01-FF for 16MB wal segments but creates 4 times less files in each "round".
Meaning we have 3F as last file after which postgres increases second number in WAL name and goes back to 01 for third one.

This in turn causes errors and failed backups in barman saying "Error: At least one WAL file is missing."
It is probably using LSN to calculate WAL filename with 16MB segment size.

Is it possible to tell barman that the configuration is such? It should probably detect this by itself.

Examples:

```
postgres=# SELECT pg_walfile_name('113/88B5ADC0');
     pg_walfile_name
--------------------------
 000000010000011300000022

postgres=# SELECT pg_walfile_name('113/FF000000');
     pg_walfile_name
--------------------------
 00000001000001130000003F
(1 row)
```

**Comment by gbartolini** [comment]
Barman is able to detect the segment size already. We will provide a fix for version 2.11.

**Comment by dsmoljanovic** [comment]
If it helps; we created local a patch for our use-case to deal with this quickly. I'm not 100% sure it covers all edge cases; but for 64MB size works fine so far. Did just manual testing.

File barman/xlog.py  function location_to_xlogfile_name_offset. Changed seg and offset calculation as follows. 
```
xlog_file_size_proper = ((xlog_file_size(xlog_segment_size) ^ 0xFFFFFFFF) +1 )
seg = (lsn & 0xFFFFFFFF) // xlog_file_size_proper
offset = (lsn & 0xFFFFFFFF) % xlog_file_size_proper
```
